llvm/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -1540,11 +1590,22 @@ int llvm::rewriteLoopExitValues(Loop *L, LoopInfo *LI, TargetLibraryInfo *TLI,
     // Replace PN with ExitVal if that is legal and does not break LCSSA.
     if (PN->getNumIncomingValues() == 1 &&
         LI->replacementPreservesLCSSAForm(PN, ExitVal)) {
+      addDebugValuesToLoopVariable(Phi.ExitBlock, ExitVal, PN);
       PN->replaceAllUsesWith(ExitVal);
       PN->eraseFromParent();
     }
   }
 
+  // If there are no PHIs to be rewritten then there are no loop live-out
+  // values, try to rewrite variables corresponding to the induction variable
```

Comment:
I reckon you want to say "rewrite debug variables" or similar, just to signal to the reader that these aren't for SSA variables which might have a functional effect on the program.

Reply:
Changed.